packages/block-library/src/post-featured-image/edit.js

```diff
@@ -67,6 +74,20 @@ function PostFeaturedImageDisplay( {
 			select( coreStore ).getMedia( featuredImage, { context: 'view' } ),
 		[ featuredImage ]
 	);
+	const mediaUrl = getMediaSourceUrlBySizeSlug( media, sizeSlug );
+
+	const imageSizes = useSelect(
+		( select ) => select( blockEditorStore ).getSettings().imageSizes,
+		[]
+	);
+	const imageSizeOptions = imageSizes
+		.filter( ( { slug } ) => {
+			return media?.media_details?.sizes?.[ slug ]?.source_url;
+		} )
+		.map( ( { name, slug } ) => ( {
+			value: slug,
+			label: name,
+		} ) );
```

Comment:
I should probably memoize this value, but I'm not 100% convinced that it won't be premature optimization.